src/scripts/notify.sh

```diff
@@ -121,22 +112,45 @@ CheckEnvVars() {
     fi
 }
 
+AbortPost() {
+    echo "NO SLACK ALERT"
+    echo
+    for s in "$@"; do
+        echo "$s"
+    done
+    exit 0
+}
+
 ShouldPost() {
-    if [ "$CCI_STATUS" = "$SLACK_PARAM_EVENT" ] || [ "$SLACK_PARAM_EVENT" = "always" ]; then
-        # In the event the Slack notification would be sent, first ensure it is allowed to trigger
-        # on this branch or this tag.
-        FilterBy "$SLACK_PARAM_BRANCHPATTERN" "${CIRCLE_BRANCH:-}"
-        FilterBy "$SLACK_PARAM_TAGPATTERN" "${CIRCLE_TAG:-}"
+    if [ "$CCI_STATUS" != "$SLACK_PARAM_EVENT" ]; then
+        if [ "$SLACK_PARAM_EVENT" != "always" ]; then
+            AbortPost \
+                "This command is set to send an alert on: $SLACK_PARAM_EVENT" \
+                "Current status: ${CCI_STATUS}"
+        fi
+    fi
 
-        echo "Posting Status"
+    # In the event the Slack notification would be sent, first ensure it is allowed to trigger
+    # on this branch or this tag.
+    if [ -n "${CIRCLE_BRANCH:-}" ]; then
+        if ! FilterBy "$SLACK_PARAM_BRANCHPATTERN" "$CIRCLE_BRANCH"; then
+            AbortPost \
+                "Branch pattern does not match: ${SLACK_PARAM_BRANCHPATTERN}" \
+                "CI was triggered by branch: ${CIRCLE_BRANCH}"
+        fi
+    elif [ -n "${CIRCLE_TAG:-}" ]; then
+        if ! FilterBy "$SLACK_PARAM_TAGPATTERN" "$CIRCLE_TAG"; then
+            AbortPost \
+                "Tag pattern does not match: ${SLACK_PARAM_TAGPATTERN}" \
+                "CI was triggered by tag: ${CIRCLE_TAG}"
+        fi
```

Comment:
Hi my Bash isn't what it once was, I'm just checking on something because I'm seeing it in my own runs.

Does your code handle the situation where there is a tag pattern but no Git tag (aka it's running from a branch). And if so, that means, since it'll fail the `tag_pattern` that no Slack posting will happen in that instance?

Reply:
It should; the `[ -n "${CIRCLE_TAG:-}" ]` conditional will be `true` only if `CIRCLE_TAG` is set